File: 프로그래머스/unrated/181864. 문자열 바꿔서 찾기/문자열 바꿔서 찾기.java
class Solution {
    public int solution(String myString, String pat) {
        int answer = 0;
      char[] charArray = myString.toCharArray();
        
        for (int i = 0; i < charArray.length; i++) {
            if (charArray[i] == 'A') {
                charArray[i] = 'B';
            } else if (charArray[i] == 'B') {
                charArray[i] = 'A';
            }
        }

        String replacedString = new String(charArray);
        
        if (replacedString.contains(pat)) {
            answer = 1;
        } else {
            answer = 0;
        }
        
        return answer;
    }
}

File: 프로그래머스/0/181864. 문자열 바꿔서 찾기/문자열 바꿔서 찾기.js
function solution(myString, pat) {
    // 'A'와 'B' 둘 다 포함되어 있는지 확인
    if (myString.includes('A') || myString.includes('B')) {
        // A와 B를 동시에 교체
        var tempString = "";
        
        for (var i = 0; i < myString.length; i++) {
            var char = myString[i];
            
            if (char === 'A') {
                tempString += 'B';
            } else if (char === 'B') {
                tempString += 'A';
            } else {
                tempString += char;
            }
        }

        myString = tempString;
    }

      return tempString.includes(pat) ? 1 : 0;
}
